### ISSUE TITLE
Reference image without image picker but by path

### ISSUE THREAD

**Issue by Sesa1988** [singerdmx/flutter-quill#2309]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/singerdmx/flutter-quill/issues)

### Use case

Hi,

My users can pick files and images via other ways and I display them outside the editor. I want to insert only these images that are already "picked" and processed by a local file path (App Directory) without an image picker. 

Is it possible to write a custom action to do that or is it supported but I didnt see it? I did not find any way to do it. 

### Proposal

An action to add a image not from an image picker but from a local path.

**Comment by EchoEllet** [comment]
Do you want to ensure that users can't copy images from other apps and paste them into your app? Or customize something? How is it being handled or only allow inserting local images?

**Comment by Sesa1988** [comment]
> Do you want to ensure that users can't copy images from other apps and paste them into your app? Or customize something? How is it being handled or only allow inserting local images?

Yes, I want to prevent everything like picking an image from the gallery, copy paste etc. 

It should only be possible to reference an image that is already in the Application Directory what comes from other flows outside the editor. 

I know the paths to these images and it would be nice if there would be an action to add an image that just expects an local path.

**Comment by Sesa1988** [comment]
> Do you want to ensure that users can't copy images from other apps and paste them into your app? Or customize something? How is it being handled or only allow inserting local images?

It would be nice without any external dependencies because I already use file picker for other parts of the app. All images are copied to the applocation context and I can access them directly. Would be amazing if we just could provide a path to an image and it would be rendered.

Its lightweight like this, can be used more flexible like the example I provided and no extension is needed. Also no dependencies issues with all the different picker options and indirect dependencies.

**Comment by Sesa1988** [comment]
I think we can close this issue. I implemented it via custom builders and it was pretty straight forward.
